src/lexer/mod.rs

```diff
@@ -119,6 +119,7 @@ impl Display for Keyword {
 pub enum TokenKind {
     Keyword(Keyword),   // reserved keywords
     Identifier(String), // [a-zA-Z](A-Za-z0-9_)*
+    Generic(String),    // [A-Z]*
```

Comment:
I think Identifier should be used here, and the uppercase (if we want to enforce it) should be enforced in the parser

Reply:
In another subsequent PR:
 - Generic is removed
 - uses Identifier